### ISSUE TITLE
Use XDG standard directories instead of hidden files in the home directory

### ISSUE THREAD

**Issue by legendre6891** [lukelbd/proplot#204]
Would it be possible to change (or, at as an option) the default locations of `~/.proplotrc` and `~/.proplot/fonts` and so on to the corresponding XDG standards?  A description may be found here: 

https://wiki.archlinux.org/index.php/XDG_Base_Directory

The standard has been gaining steam in the past five or so years; the idea is that apps don't have to litter the home directory with dot files, which makes backing up configuration files easier (not to mention, it keeps the home directory tidy!).

The TLDR is that: 

- The file `~/.proplotrc` should be moved to `~/.config/proplot/proplotrc`
- The folders `~/.proplot/*` should be moved to `~/.config/proplot/*`

(Actually, `~/.config` is more generally the value of the environment variable `XDG_CONFIG_HOME`.)

Happy to submit a PR if there is interest.

**Comment by lukelbd** [comment]
Thought this issue would pop up eventually. Was aware of the standard, but I avoided it basically out of laziness and because I'm used to dumping everything in `$HOME`... I guess lazy developers are the main impediment to this standard being adopted ;)

We could address this by [going the way of vim](https://github.com/vim/vim/issues/2034): permit both `~/.proplot` and `$XDG_CONFIG_HOME/proplot` for the configuration folder, and permit all of `~/.proplotrc`, `~/.proplot/proplotrc`, and `$XDG_CONFIG_HOME/proplot/proplotrc` for the configuration file. We can issue a warning message when a `proplotrc` file is found in more than one of these locations, but still read all of them. A *default* proplotrc file will be generated in the XDG standard location only if all 3 of these paths are empty.

For consistency, we should also detect *local* files named `proplotrc` (without the leading dot), but we can continue to detect `.proplotrc` files for backwards compatibility. There already seems to be some precedent for allowing both hidden and non-hidden versions of configuration files ([e.g. the readthedocs configuration](https://docs.readthedocs.io/en/stable/config-file/index.html)).

This isn't an immediate priority but happy to accept PRs in the meantime. The relevant code is here:

https://github.com/lukelbd/proplot/blob/abdca188a7588d35f81b1362fbdbbe258c84a6df/proplot/config.py#L509-L532

The default proplotrc file is generated here:

https://github.com/lukelbd/proplot/blob/abdca188a7588d35f81b1362fbdbbe258c84a6df/proplot/config.py#L1529-L1533

**Comment by legendre6891** [comment]
Thank you, Luke; your thoughts and pointers are very helpful. I'm a bit confused about this part:

> For consistency, we should also detect local files named proplotrc (without the leading dot), but we can continue to detect .proplotrc files for backwards compatibility.  There already seems to be some precedent for allowing both hidden and non-hidden versions of configuration files (e.g. the readthedocs configuration).

I took a look at the readthedocs link, but the use case there seems to be that you can put a configuration file in the folder. Is what you mean that proplot should also look in the current directory, i.e. `$PWD/proplotrc` and `$PWD/.proplotrc` (for some suitable value of `$PWD`)? Or do you simply mean that we should also look at `$HOME/proplotrc`? 

Cheers!

**Comment by lukelbd** [comment]
No problem! Yep, I just meant that when proplot is looking for "loose" proplotrc files (including `$HOME/proplotrc`, `$PWD/proplotrc`, `$PWD/../proplotrc`, etc.), it should also look for the hidden filename version  `.proplotrc`. And the justifications for supporting both going forward are:

1. Backwards compatibility: Older proplot versions supported only hidden `.proplotrc` files
2. Supporting both hidden/unhidden file names for "loose" configuration files has precedent, because both approaches have advantages -- hidden files are less obtrusive, but unhidden files are more obvious and less likely to surprise the user.

For proplotrc files inside `proplot` and `.proplot` subfolders, I suppose there is no reason to support the hidden `.proplotrc` version, since the config file is not obtrusive (it is sequestered in a separate subfolder). But I can see the argument for making `.proplotrc` and `proplotrc` synonymous everywhere (less confusing for users). Ambivalent on this.

There should probably be a public `RcConfigurator.config_paths` method or similar that shows the location of all proplotrc files observed by proplot, but that might be a separate issue/PR.

**Comment by lukelbd** [comment]
I haven't implemented this yet, but in #204 I added support for `proplotrc` files (without the leading dot) because this was really straightforward to add and I think "public"/unhidden configuration files is probably what most people prefer.

**Comment by lukelbd** [comment]
This is now fully implemented on master. The system-dependent user `proplotrc` file and `.proplot` folder locations are given by `pplt.rc.user_file()` and `pplt.rc.user_folder()`. Also `~/.proplotrc` and `~/.proplot` are accepted on all systems for backwards compatibility. If multiple valid files or folders are found a warning is raised.

I determined suitable default locations mostly be mimicking what matplotlib does for `matplotlibrc`. On most systems the new defaults are `~/.proplot` and `~/.proplot/proplotrc`. On Linux they are in `~/.config` or in `$XDG_CONFIG_HOME` unless those alternative/universal locations already exist.
